### DISCUSSION TITLE
How to use with junstyle php-cs-fixer VS Code extension?

### DISCUSSION THREAD

**Discussion by tanc** [drupol/phpcsfixer-configs-drupal#15] [Q&A] [answered]
# Summary of problem

I'd like to use php-cs-fixer in Visual Studio Code to format Drupal code. This looks like the only rulesets available that might work but I'm struggling to understand how to get php-cs-fixer extension to read the rulesets. Everything I've read says there should be a `php_cs` or `php_cs.dist` file but there is none in your repo.

## Steps required to reproduce the problem

1. Install junstyle php-cs-fixer extension in VS Code
2. Set php-cs-fixer as the default formatter for PHP files
3. Attempt to install this repo globally and refer to it from the `php-cs-fixer.rules` setting

## Expected Result

I'm hoping these rules will somehow be read

## Actual Result

I've no idea how to set this up.

**Comment by aprice42** [comment]
@tanc I know it has been some time since this post was made.. did you make any progress on this? I am also trying to get Drupal standards to work with this VS code plugin. Any advice would be appreciated.

**Comment by WengerK** [comment, marked as answer]
Do any of you have a solution to assist the community with this discussion?
Have you found a working solution for using PHP-CS-Fixer with VSCode?

In my opinion, here's how to make VSCode work seamlessly with PHP-CS-Fixer, especially when using the Drupal rule set:

1. Start by installing a VSCode extension that utilizes PHP-CS-Fixer from your project installation.

2. Add PHP-CS-Fixer and PHP-CS-Fixer Configs Drupal to your Drupal project by running the following command:

```bash
composer require --dev drupol/phpcsfixer-configs-drupal
```

3. Create a `.php-cs-fixer.php` file within your Drupal project, using the recommended Drupal rule set provided by this repository:

```php
# .php-cs-fixer.php
<?php

use drupol\PhpCsFixerConfigsDrupal\Config\Drupal8;

$finder = PhpCsFixer\Finder::create()
  // Specify the directory you want to lint.
  ->in(['web/modules/custom'])
  ->name('*.module')
  ->name('*.inc')
  ->name('*.install')
  ->name('*.test')
  ->name('*.profile')
  ->name('*.theme')
  ->notPath('*.md')
  ->notPath('*.info.yml')
;

// Choose between Drupal8+ or Drupal7 rule sets here.
$config = new Drupal8();
$config->setFinder($finder);

// You can customize rules to match your code style if needed.
$rules = $config->getRules();
$rules['global_namespace_import'] = FALSE;
$rules['no_superfluous_phpdoc_tags'] = FALSE;
$rules['ordered_class_elements']['sort_algorithm'] = 'none';

$config->setRules($rules);
return $config;
```

4. Run PHP-CS-Fixer locally to ensure it works as a standalone tool. Then, configure your VSCode extension to use it:

```
./vendor/bin/php-cs-fixer fix --config=.php-cs-fixer.php
```
